pulp_rpm/app/tasks/publishing.py

```diff
@@ -247,12 +247,9 @@ def get_checksum_type(name, checksum_types):
     """
     original = checksum_types.get("original")
     metadata = checksum_types.get("metadata")
-    checksum_type = original.get(name, CHECKSUM_TYPES.SHA256)
+    checksum_type = metadata if metadata else original.get(name, CHECKSUM_TYPES.SHA256)
 
-    if metadata:
-        checksum_type = metadata
-
-    return getattr(cr, getattr(CHECKSUM_TYPES, checksum_type.upper()), cr.SHA256)
```

Comment:
Splitting this apart to demonstrate what happened:

```python
# checksum_type = "sha384"
pulp_checksum_enum_name = checksum_type.upper()  # "SHA384"
converted_checksum = getattr(CHECKSUM_TYPES, pulp_checksum_enum_name)  # "sha384"
return getattr(cr, converted_checksum, cr.SHA256)  # no attribute "sha384", so it falls back to the default
```

Reply:
Put this directly into the code as a comment, to save the sanity of Future Maintainers (who might be you :) )